### ISSUE TITLE
Mozilla's Fluent Message Format Support

### ISSUE THREAD

**Issue by chrischen** [lingui/js-lingui#489]
**Is your feature request related to a problem? Please describe.**
Mozilla's new Fluent format is similar to MessageFormat from ICU but supports asymmetric localization (where the source and target language don't have a different number of possible translations depending on a variable or number or context).

```
tabs-close-warning-multiple = {$count ->
    [few] Chystáte se zavřít {$count} panely. Opravdu chcete pokračovat?
   *[other] Chystáte se zavřít {$count} panelů. Opravdu chcete pokračovat?
}
```

**Describe the solution you'd like**
Consider the support of Fluent formatted translation files: https://hacks.mozilla.org/2019/04/fluent-1-0-a-localization-system-for-natural-sounding-translations/

**Describe alternatives you've considered**

**Additional context**

**Comment by tricoder42** [comment]
I'm in favor of supporting Fluent, but I don't have capacity to work on it.

- `@lingui/macro` was refactored, that the code is parsed into tokens and message is generated from these tokens. Right now the only implemented message format is [ICU](https://github.com/lingui/js-lingui/blob/next/packages/macro/src/icu.ts). You could implement Fluent in the same way.

- `@lingui/conf` probably needs another option, `format`:

```js
{
  parser: "icu" | "fluent"
}
```

- `@lingui/cli` and `@lingui/loader` - once we have configuration in `conf`, we could compile messages using specific parser

Bottom line: Packages are ready for another message format. We just need somebody to put the effort into it. Interested in sending a PR? 😉

**Comment by nodkz** [comment]
I'm working on FLUENT integration right now.

For now implemented `format: fluent` for `@lingui/cli` and `@lingui/loader`: https://github.com/nodkz/js-lingui/commit/f38a9af95bc975d0f209eba6c1c818058c69905b

It remains to implement new `parser` option.

PS. @tricoder42 please review https://github.com/lingui/js-lingui/pull/530 I don't know why tests fail cause it perfectly works locally. This PR 530 is required for proper testing of `fluent` formatter and implemented on top of this changes. Tnx.

**Comment by nodkz** [comment]
Ouch, `parser` implemented in the `next` branch.
Ok, I'll port my changes to this branch.

@tricoder42 you may close #530 it not required anymore.

**Comment by alesn** [comment]
This is still relevant ...

**Comment by tricoder42** [comment]
@alesn Are you going to work on this?

**Comment by tricoder42** [comment]
Closing as a stale issue. Feel free to reopen when you start working on this or send PR right away 👍
